foundation/org.eclipse.persistence.core.test.framework/src/main/java/org/eclipse/persistence/testing/tests/performance/emulateddb/EmulatedDatabaseMetaData.java

```diff
@@ -0,0 +1,913 @@
+/*
+ * Copyright (c) 2024 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+// Contributors:
+//     Oracle - initial API and implementation
+package org.eclipse.persistence.testing.tests.performance.emulateddb;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.RowIdLifetime;
+import java.sql.SQLException;
+
+public class EmulatedDatabaseMetaData implements DatabaseMetaData {
+
+    private EmulatedConnection connection;
+
+    public EmulatedDatabaseMetaData(EmulatedConnection emulatedConnection) {
```

Comment:
this should probably be a `Connection` instead and where the cast is needed, one should call `unwrap`, WDYT?

Reply:
I'm not sure if it helps with anything except little bit complicated code. This is our implementation of `DatabaseMetaData` and almost any access to `this.connection` will need cast/unwrap call as specific properties are fetched there.

Reply:
...and forcing us to use the right API calls in tests... but I'll leave it up to you ;-)